src/Elastic.Apm/Metrics/MetricsProvider/GcMetricsProvider.cs

```diff
@@ -68,54 +58,18 @@ public GcMetricsProvider(IApmLogger logger, IReadOnlyList<WildcardMatcher> disab
 			_collectGcGen1Size = !WildcardMatcher.IsAnyMatch(disabledMetrics, GcGen1SizeName);
 			_collectGcGen2Size = !WildcardMatcher.IsAnyMatch(disabledMetrics, GcGen2SizeName);
 			_collectGcGen3Size = !WildcardMatcher.IsAnyMatch(disabledMetrics, GcGen3SizeName);
-
-			if (!IsEnabled(disabledMetrics))
-				return;
+			_isEnabled = _collectGcCount || _collectGcTime || _collectGcGen0Size || _collectGcGen1Size || _collectGcGen2Size || _collectGcGen3Size;
+			if (!IsEnabled(disabledMetrics)) return;
 
 			_logger = logger.Scoped(DbgName);
-
-			if (PlatformDetection.IsDotNetFullFramework)
+			if (!PlatformDetection.IsDotNetCore && !PlatformDetection.IsDotNet)
 			{
-				// if the OS doesn't support filtering, the processes and runtimes tracked by the trace event session source can grow
-				// unbounded, leading to the application consuming ever more memory.
-				if (!TraceEventProviderOptions.FilteringSupported)
-				{
-					_logger.Info()?.Log("TraceEventSession does not support filtering on this operating system. GC metrics won't be collected");
-					return;
-				}
-
-				try
-				{
-					TraceEventSessionName = SessionNamePrefix + Guid.NewGuid();
```

Comment:
`SessionNamePrefix` still defined above (can't add suggestion, because that line is unchanged). We should remove it.